lib/evaluator.py

```diff
@@ -4,6 +4,7 @@
 and evaluates the given recommender.
```

Comment:
In general, ndcg, mrr, and recall_at_x methods should be combined into one. The computationally expensive part is calculating top recommendations and they are always the same for every user so it doesn't make sense to do it 3 times. If you opt to write a helper method that calculates top recommendations for every user and passes it to the 3 methods, you will add o(kn) space complexity. So can we merge them in one method please?